cloud/scope/clients_test.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scope
+
+import (
+	"os"
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestGettingEnvironment(t *testing.T) {
```

Comment:
what happens when `AZURE_ENVIRONMENT` is an invalid cloud name?
 eg. If I set `AZURE_ENVIRONMENT` to `AzureUAGovernmentCloud`  I should get an error rather than CAPZ trying to use PublicCloud as default and failing later because my sub isn't valid in public cloud.

Reply:
added a test for an invalid value, good catch